### DISCUSSION TITLE
Sloeber 4.4.1 compile error for ESP32 Dev Board

### DISCUSSION THREAD

**Discussion by cmburns** [Sloeber/arduino-eclipse-plugin#1584] [Q&A]
Hi,

today I installed Sloeber 4.4.1 on my Windows 11 machine.
I installed the ESP32 platform (https://raw.githubusercontent.com/espressif/arduino-esp32/gh-pages/package_esp32_index.json).
Then I wanted to start with simple Blink example for my ESP32 WROOM. The board I used is 'ESP32 Dev Module'.
But I get following compile error:

`...
DARDUINO_BOARD=\"ESP32S3_DEV\"" -DARDUINO_VARIANT="esp32s3" -DARDUINO_PARTITION_default  -DESP32 -DCORE_DEBUG_LEVEL=0 -DARDUINO_RUNNING_CORE=1 -DARDUINO_EVENT_RUNNING_CORE=1  -DARDUINO_USB_MODE=1 -DARDUINO_USB_CDC_ON_BOOT=0 -DARDUINO_USB_MSC_ON_BOOT=0 -DARDUINO_USB_DFU_ON_BOOT=0 "@C:\Users\Thomas\Documents\sloeber441workspace\Test\Release/file_opts"   -I"C:\Sloeber441\arduinoPlugin\packages\esp32\hardware\esp32\2.0.14\variants\esp32s3" -I"C:\Sloeber441\arduinoPlugin\packages\esp32\hardware\esp32\2.0.14\cores\esp32" -MMD -MP -MF"sloeber.ino.cpp.d" -MT"sloeber.ino.cpp.o" -D__IN_ECLIPSE__=1 -x c++ "..\sloeber.ino.cpp"   -o "sloeber.ino.cpp.o"
**xtensa-esp32s3-elf-g++: error: C:\Users\Thomas\Documents\sloeber441workspace\Test\Release/file_opts: No such file or directory**
make[1]: *** [subdir.mk:20: sloeber.ino.cpp.o] Error 1
make: *** [makefile:97: all] Error 2
"C:/Sloeber441/arduinoPlugin/tools/make/make all" terminated with exit code 2. Build might be incomplete.

12:20:24 Build Failed. 2 errors, 0 warnings. (took 2s.115ms)
`

The blink.ino looks like this:

`// the setup function runs once when you press reset or power the board
void setup() {
  // initialize digital pin LED_BUILTIN as an output.
  pinMode(LED_BUILTIN, OUTPUT);
}

// the loop function runs over and over again forever
void loop() {
  digitalWrite(LED_BUILTIN, HIGH);   // turn the LED on (HIGH is the voltage level)
  delay(1000);                       // wait for a second
  digitalWrite(LED_BUILTIN, LOW);    // turn the LED off by making the voltage LOW
  delay(1000);                       // wait for a second
}
`

In the IDE I get red underlines, if I hover the mouse it says:

Function pinMode could not be resolved
Function delay could not be resolved
Function digotalWrite could not be resolved

and further:

Symbol LED_BUILTIN could not be resolved
Symbol OUTPUT could not be resolved
Symbol HIGH could not be resolved
Symbol LOW could not be resolved

**What I am missing please?**

**Comment by jantje** [comment]
This is a issue about ESP32 using a undocumented feature https://github.com/arduino/arduino-cli/issues/2369
See issue #1582 on how to work around it

**Comment by cmburns** [comment]
@jantje: OK first issue regarding file_opts is solved by adding two lines in platform.txt and following the [#1533](https://github.com/Sloeber/arduino-eclipse-plugin/discussions/1533)

But I still have the '... could not be resolved' errors at hand.
What I wonder is, I can compile the blink.ino without errors and it runs fine on my ESP32.

But how to get rid of all those '...could not be resolved' errors in the IDE please?

**Comment by jantje** [comment]
Check the sloeber tutorial: https://github.com/Sloeber/arduino-eclipse-plugin/discussions/1555
Specifically this section: https://github.com/ArminJo/SloeberTutorial#strange-errors-indicated-f3-does-not-work-index-not-complete

**Comment by cmburns** [comment]
Hi,

thanks, I already right clicked the project and chose 'Freshen all files'. But it seems nothing happens inside the IDE.
The red underlines are still there and the messages 'Function XYZ could not be resolved' as well.

Am I still missing something?

**Comment by jantje** [comment]
>Am I still missing something?

The indexer (Which is a CDT thing not Sloeber thing) is picky and sometimes the indexer just doesn't want to do it.
Opening and closing the workspace sometimes helps.
Checking for error logs
Trying the other options (like rebuild)

**Comment by cmburns** [comment]
Starting the IDE new and Index-->Rebuild did help - thanks!
